### ISSUE TITLE
Suggestion: multistage Docker builds

### ISSUE THREAD

**Issue by weishiuchang** [travisghansen/external-auth-server#71]
Currently the docker image for `external-auth-server` sits at just over 1.04GB. This can be reduced using multi-staged builds. Here's an initial pass at a Dockerfile for eas:

```
FROM node:10 AS builder

WORKDIR /root
COPY package*.json ./
RUN npm install

FROM node:10-alpine AS release

# Run as a non-root user
RUN adduser --disabled-password eas \
        && mkdir /home/eas/app \
        && chown -R eas: /home/eas
WORKDIR /home/eas/app
USER eas

COPY --from=builder --chown=eas:eas /root/node_modules ./node_modules
COPY --chown=eas:eas . .

EXPOSE 8080

CMD [ "npm", "start" ]
```

Using node:10-alpine the resultant image ends up being just under 200MB:

```bash
docker images external-auth-server:slim
REPOSITORY                           TAG                 IMAGE ID            CREATED             SIZE
external-auth-server                 slim              bfc66bf5f94c        38 seconds ago      194MB
```

I haven't tested to see if eas will work with this change, but I thought I'd at least suggest a way to reduce the current eas image by 4/5ths.

**Comment by travisghansen** [comment]
Yes! I love it! I need to merge my `next` branch first as I moved to node 12. After that we can revisit this. I want to say I looked into multistage early on and couldn't use it. Hopefully that's no longer the case.

**Comment by travisghansen** [comment]
I have this implemented and will merge to master the next time I do a release. Images are ~80MB (compressed in docker hub) so much improved size reduction.

**Comment by Electrofenster** [comment]
could you test it with the `node:12.20.2-alpine3.10` (28.6 MB) image?
On docker hub this image is a bit smaller than the `node:12-buster-slim` (51.5 MB) image so we could save a few mb ;)

**Comment by travisghansen** [comment]
For binary compatibility with some pre-built npm stuff I think we’ll stick with glibc over musl based system.

**Comment by travisghansen** [comment]
Implemented with `v0.10.0`.
